client/asset/eth/eth.go

```diff
@@ -296,6 +297,28 @@ func (eth *ExchangeWallet) Balance() (*asset.Balance, error) {
 	return bal, nil
 }
 
+// getInitGas gets an estimate for the gas required for initiating a swap.
+func (eth *ExchangeWallet) getInitGas() (uint64, error) {
+	now := time.Now().Unix()
+	var secretHash [32]byte
+	copy(secretHash[:], encode.RandomBytes(32))
+	parsedAbi, err := abi.JSON(strings.NewReader(swap.ETHSwapABI))
+	if err != nil {
+		return 0, err
+	}
+	data, err := parsedAbi.Pack("initiate", big.NewInt(now), secretHash, &eth.acct.Address)
```

Comment:
Is there reason to believe we need to use anything but `math.NewInt(1)` and `[32]byte{1}`?  Although we won't be calling `getInitGas` on a hot path, the system clock and crypto/rand are relatively expensive and it's not clear we need to use them at all.

Reply:
I guess the only real reason is if someone initiates a swap with secret key 1, then it will break gas estimation for everyone.

Reply:
Ahhh, so the idea is this secret hash must be unused by the contract on-chain so that it follows the expected execution path and thus gas estimate?

Reply:
Yes. I'll can definitely change the use of system clock though.